src/main/java/org/candlepin/insights/controller/InventoryController.java

```diff
@@ -279,8 +279,16 @@ public OrgInventory getInventoryForOrg(String orgId) {
     }
 
     private Stream<ConduitFacts> validateConduitFactsForOrg(String orgId) {
-        return StreamSupport.stream(pinheadService.getOrganizationConsumers(orgId).spliterator(), false)
-            .map(this::validateConsumer).filter(Optional::isPresent).map(Optional::get);
+        List<ConduitFacts> conduitFacts = new ArrayList<>();
+        for (Consumer consumer : pinheadService.getOrganizationConsumers(orgId)) {
+            // If a consumer is missing an accountNumber, it means they all are and we should just stop
+            if (StringUtils.isEmpty(consumer.getAccountNumber())) {
+                log.warn("Account number is missing on consumers from {}", orgId);
+                break;
+            }
+            validateConsumer(consumer).ifPresent(conduitFacts::add);
+        }
+        return conduitFacts.stream();
```

Comment:
hrmm... Now I see the challenge you were having with this. Don't we now end up iterating the consumer list twice when we do this when all records have an account? Could we instead put a peek method on the PagedConsumerIterator so that the first thing we do in validateConduitFactsForOrg(...) is peek at the first record after the lookup and check for the account. If not found, just return an empty stream after logging. Otherwise, return the Stream as it originally was created.

Rough  likely not working code:

```java
PagedConsumerIterator consumerIter = pinheadService.getOrganizationConsumers(orgId);
if (StringUtils.isEmpty(consumerIter.peek().getAccountNumber()) {
     return new ArrayList().stream();
}
return StreamSupport.stream(consumerIter.spliterator(), false)
            .filter(Optional::isPresent).map(Optional::get);
```

Reply:
I was a little torn about this.  The way I have it, we check every consumer for a missing account number.  If we find one, we break immediately.  So for the expected bad case (the very first consumer has a missing account number), we break almost immediately.  For the expected good case, we do N checks and then N adds to a list and then stream the list.  Not great but it's paranoid about consumers not being uniform on missing accountNumbers.

With your suggestion we just check the first one and then if that passes we send everything through.  Given that we have been explicitly told "if the account number is missing on one, it will be missing on all", I think this is a better approach.